components/script/dom/document.rs

```diff
@@ -574,6 +578,28 @@ impl Document {
         }
     }
 
+    /// https://html.spec.whatwg.org/multipage/#scroll-to-the-fragment-identifier
+    pub fn check_and_scroll_fragment(&self, fragment: &str) {
+        let body = self.GetBody();
+        let target = self.find_fragment_node(fragment);
+
+        // Step 1
+        self.set_target_element(target.r());
+
+        let target = target.r().or_else(||
+                                        // Step 2
+                                        if fragment.is_empty() {
+                                            body.r().map(|b| b.upcast::<Element>())
```

Comment:
So the origin of the stacking context space where all bounding box live in is: `                LogicalPoint::zero(writing_mode).to_physical(writing_mode, viewport_size)`, which is not (0, 0) when the writing mode is not `horizontal-lr`. But currently servo does not care about writing mode too much, So I will simply use (0, 0) to scroll to the top.